AsyncDisplayKit/ASImageNode.mm

```diff
@@ -460,8 +464,12 @@ + (UIImage *)createContentsForkey:(ASImageNodeContentsKey *)key isCancelled:(asd
   // Another option is to have ASDisplayNode+AsyncDisplay coordinate these cases, and share the decoded buffer.
   // Details tracked in https://github.com/facebook/AsyncDisplayKit/issues/1068
   
-  @synchronized(key.image) {
-    [key.image drawInRect:key.imageDrawRect];
+  // We want to use copy, but if our image has alpha and our context is dirty, we have to use normal.
+  UIImage *image = key.image;
+  CGBlendMode blendMode = contextIsClean == NO && CGImageGetAlphaInfo(image.CGImage) != kCGImageAlphaNone ? kCGBlendModeNormal : kCGBlendModeCopy;
```

Comment:
@Adlai-Holler This is a great patch, thank you!

This line is too complex to read.  If you leave it compound, please put parenthesis around the two elements before the ?.

Probably better is to create a BOOL imageHasAlpha or imageIsOpaque.